testsV2/ut_repyv2api_getresourcescopieslimitsdict.r2py

```diff
@@ -0,0 +1,33 @@
+"""Ensure that the `getresources` call returns a copy of the `limits` 
+dict (and not the original dict), thereby ensuring that user code can 
+inspect but not modify the resource restrictions that `nanny.py` enforces.
+
+We check this by trying to add a new, previously disallowed `connport` 
+and then using it. If the port is still not allowed, all is fine.
+"""
+#pragma repy
+limits, usage, stoptimes = getresources()
+
+# This is the set of allowed TCP ports. (We should see only a *copy* 
+# of the set that `nanny` uses.)
+allowed_ports = limits["connport"]
+
+
+# Try to add an unprivileged port that wasn't allowed before
+for new_port in xrange(1024, 65536):
+  if new_port not in allowed_ports:
+    allowed_ports.add(new_port)
+    break
+else:
+  log("Premature error: All unprivileged ports are allowed already!\n")
+
+
+# Verify that we are still not allowed to use the newly added port
+try:
+  listen_socket = listenforconnection("127.0.0.1", new_port)
+except ResourceForbiddenError:
+  # This is the expected, correct error. (Any other exceptions are neither.)
+  pass
+else:
+  log("Bad: User code succeeded adding a port to the set of allowed connports and use it!\n")
+
```

Comment:
Ran unit test before and after applying fix for repy wrapped getresources function override 48aa0a0b8b18de89d6e3e2b1cc6364dedcce4de4. 

Before the fix the unit test fails, i.e. the user is able to add a previously not allowed port to the port list and then also this port. 

After the fix the test passes, i.e. the user can still add a not allowed port, but cannot use it, because, s/he adds it to a copy of the port list.

Also verified that allowed ports can still be used.

*Note: another way of testing the fix would be to add a port, then call `getresources` again and see if the port was actually added*